tests/beman/inplace_vector/compare.test.cpp

```diff
@@ -0,0 +1,181 @@
+#include <beman/inplace_vector/inplace_vector.hpp>
+#include <gtest/gtest.h>
+
+#include <compare>
+
+using namespace beman;
+
+template <class T>
+concept has_threeway = requires(const T &t) {
+  { t <=> t };
+};
+
+TEST(Compare, threeway_int) {
+  using T = inplace_vector<int, 4>;
+  T empty{};
+  T base{1, 2, 3};
+  T copy = base;
+
+  T greater{4, 5, 6};
+  T lesser{0, 0, 0};
+
+  T bigger{1, 2, 3, 0};
+  T smaller{1, 2};
+
+  T greater_smaller{2, 2};
+  T lesser_bigger{0, 2, 3, 0};
+
+  EXPECT_TRUE((base <=> copy) == 0);
+  EXPECT_TRUE((base <=> greater) < 0);
+  EXPECT_TRUE((base <=> lesser) > 0);
+
+  EXPECT_TRUE((base <=> bigger) < 0);
+  EXPECT_TRUE((base <=> smaller) > 0);
+
+  EXPECT_TRUE((base <=> greater_smaller) < 0);
+  EXPECT_TRUE((base <=> lesser_bigger) > 0);
+}
```

Comment:
maybe also check some synthesized operators as well? 

```
EXPECT_TRUE(smaller < bigger );
EXPECT_TRUE(bigger > smaller );
EXPECT_TRUE(base == copy );
EXPECT_TRUE(base >= copy );
EXPECT_TRUE(copy >= base );
EXPECT_TRUE(copy <= base );

Reply:
I'd like to see some tests because the synthesized comparisons early on did some odd things and created a bunch of issues -- the committee had to fix that post-hoc. So it's possible there's still some older compilers that haven't implemented -- although it's quite possible these aren't impacted.  

If you want to keep the tests smaller I'd suggest removing the operator<=> tests and only testing the actual comparisons users would call.